ts/DataGrid/DataGrid.ts

```diff
@@ -339,11 +353,22 @@ class DataGrid implements DataEventEmitter<DataGrid.Event> {
         const rowCount = this.getRowCount();
 
         for (let j = 0; j < this.rowElements.length && i < rowCount; j++, i++) {
-            const cellElements = this.rowElements[j].childNodes;
+            const rowElement = this.rowElements[j];
+            rowElement.dataset.rowIndex = String(i);
```

Comment:
@oysteinmoseng any a11y concerns regarding the dataset approach?

Reply:
All good!